lib/brainstem/dsl/field.rb

```diff
@@ -38,7 +38,12 @@ def optional?
         options[:optional]
       end
 
+      # Please override in sub classes to compute value of field with the given arguments.
```

Comment:
Interesting. So if people don't override this, it's likely to be wrong?

Reply:
The base class in this case `DSL::Field` needs to evaluated with the default `evaluate_value_on`. It's not necessary to override it but in the case of `DSL::BlockField` we override the method to make it behave differently